File: backend/models/Payment.js
const mongoose = require('mongoose');

const paymentSchema = new mongoose.Schema({
  booking: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Booking',
    required: true
  },
  customer: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  amount: {
    type: Number,
    required: true,
    min: 0
  },
  paymentMethod: {
    type: String,
    required: true,
    enum: ['Card', 'UPI', 'Cash', 'Wallet', 'Razorpay']
  },
  transactionId: {
    type: String,
    unique: true,
    sparse: true
  },
  razorpayOrderId: {
    type: String,
    sparse: true
  },
  razorpayPaymentId: {
    type: String,
    sparse: true
  },
  razorpaySignature: {
    type: String,
    sparse: true
  },
  status: {
    type: String,
    enum: ['Pending', 'Completed', 'Failed', 'Refunded'],
    default: 'Pending'
  },
  cardDetails: {
    last4Digits: String,
    cardType: String
  },
  upiId: String,
  paymentDate: {
    type: Date,
    default: Date.now
  },
  refundAmount: {
    type: Number,
    default: 0
  },
  refundDate: Date,
  notes: String,
  createdAt: {
    type: Date,
    default: Date.now
  }
});

module.exports = mongoose.model('Payment', paymentSchema);

File: backend-java/src/main/java/com/carrental/model/Payment.java
package com.carrental.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import java.util.Date;
import java.util.Map;

@Data
@NoArgsConstructor
@Document(collection = "payments")
public class Payment {

    @Id
    private String id;
    
    private String bookingId; // Booking ObjectId
    
    private String customerId; // User ObjectId
    
    private Double amount;
    
    private String paymentMethod; // Card, UPI, Cash, Wallet, Razorpay
    
    private String transactionId;
    
    private String razorpayOrderId;
    private String razorpayPaymentId;
    private String razorpaySignature;
    
    private String status = "Pending"; // Pending, Completed, Failed, Refunded
    
    private Map<String, String> cardDetails; // E.g., { last4Digits: "1234", cardType: "Visa" }
    
    private String upiId;
    
    private Date paymentDate = new Date();
    
    private Double refundAmount = 0.0;
    
    private Date refundDate;
    
    private String notes;
    
    private Date createdAt = new Date();
}
